### ISSUE TITLE
cifar10_estimator code is half as fast as the old multi-gpu code under cifar10

### ISSUE THREAD

**Issue by alawami** [tensorflow/models#4090]
### System information
- **What is the top-level directory of the model you are using**: cifar10 and cifar10_estimator
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.4.1, 1.5.0 (Docker), 1.7.0 (Docker)
- **Bazel version (if compiling from source)**: Pre-compiled and compiled Bazel (0.5.4 for TF 1.4.1 and 0.10.1 for others)
- **CUDA/cuDNN version**: 8/6 AND 9/7
- **GPU model and memory**: I have tried on the following cards: nVidia Quadro P6000, GTX 1080 Ti, Memory: 128G
- **Exact command to reproduce**:
python cifar10_main.py --data-dir=${PWD}/cifar-10-data \
                        --job-dir=/tmp/cifar10 \
                        --num-gpus=2 \ # For both 1 and 2
                        --train-steps=1000

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem

The cifar10_estimator code (https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10_estimator) runs less than half as fast as the old code (https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10).

Another issue, is while with cifar10 code the GPU can handle a batch size of 30,000, with the cifar10_estimator code it can only handle around a batch size of 6800 on a single GPU.

### Source code / logs

Old code logs:

python cifar10_multi_gpu_train.py --max_steps=1000 --num_gpus=1 --batch_size=256
Filling queue with 20000 CIFAR images before starting to train. This will take a few minutes.
2018-04-25 13:29:00.142846: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2018-04-25 13:29:00.631243: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties: 
name: Quadro P6000 major: 6 minor: 1 memoryClockRate(GHz): 1.645
pciBusID: 0000:03:00.0
totalMemory: 23.87GiB freeMemory: 23.70GiB
2018-04-25 13:29:01.008015: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 1 with properties: 
name: Quadro P400 major: 6 minor: 1 memoryClockRate(GHz): 1.2525
pciBusID: 0000:04:00.0
totalMemory: 1.94GiB freeMemory: 1.06GiB
2018-04-25 13:29:01.369006: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 2 with properties: 
name: Quadro P6000 major: 6 minor: 1 memoryClockRate(GHz): 1.645
pciBusID: 0000:84:00.0
totalMemory: 23.87GiB freeMemory: 23.70GiB
2018-04-25 13:29:01.369084: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Device peer to peer matrix
2018-04-25 13:29:01.369112: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1051] DMA: 0 1 2 
2018-04-25 13:29:01.369121: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 0:   Y N N 
2018-04-25 13:29:01.369127: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 1:   N Y N 
2018-04-25 13:29:01.369133: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 2:   N N Y 
2018-04-25 13:29:01.369145: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Quadro P6000, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-04-25 13:29:01.369154: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1107] Ignoring gpu device (device: 1, name: Quadro P400, pci bus id: 0000:04:00.0, compute capability: 6.1) with Cuda multiprocessor count: 2. The minimum required count is 8. You can adjust this requirement with the env var TF_MIN_GPU_MULTIPROCESSOR_COUNT.
2018-04-25 13:29:01.369162: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:1) -> (device: 2, name: Quadro P6000, pci bus id: 0000:84:00.0, compute capability: 6.1)
2018-04-25 13:29:04.852332: step 0, loss = 4.68 (103.5 examples/sec; 2.473 sec/batch)
2018-04-25 13:29:05.704067: step 10, loss = 4.60 (11601.0 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:05.925013: step 20, loss = 4.45 (10416.7 examples/sec; 0.025 sec/batch)
2018-04-25 13:29:06.148957: step 30, loss = 4.65 (11282.2 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:06.374599: step 40, loss = 4.41 (11980.4 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:06.600055: step 50, loss = 4.24 (11903.7 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:06.828994: step 60, loss = 4.20 (11451.0 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:07.051775: step 70, loss = 4.28 (12756.2 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:07.275746: step 80, loss = 4.30 (11583.0 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:07.497551: step 90, loss = 4.13 (11597.9 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:07.730825: step 100, loss = 4.11 (10949.7 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:08.373678: step 110, loss = 3.97 (11677.7 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:08.597367: step 120, loss = 3.91 (10693.8 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:08.821110: step 130, loss = 3.94 (12464.8 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:09.044257: step 140, loss = 3.91 (11598.1 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:09.265448: step 150, loss = 3.90 (11529.4 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:09.493998: step 160, loss = 3.88 (11761.6 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:09.708620: step 170, loss = 3.88 (12573.5 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:09.942975: step 180, loss = 3.93 (9884.2 examples/sec; 0.026 sec/batch)
2018-04-25 13:29:10.182471: step 190, loss = 4.06 (12075.1 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:10.413786: step 200, loss = 3.64 (11153.8 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:11.047037: step 210, loss = 3.79 (11193.8 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:11.276044: step 220, loss = 3.81 (11493.2 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:11.499771: step 230, loss = 3.81 (12144.2 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:11.724003: step 240, loss = 3.65 (11575.7 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:11.949049: step 250, loss = 3.72 (11360.3 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:12.169903: step 260, loss = 3.66 (12675.1 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:12.395107: step 270, loss = 3.59 (11294.1 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:12.622525: step 280, loss = 3.49 (11322.3 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:12.846273: step 290, loss = 3.70 (11206.1 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:13.070948: step 300, loss = 3.58 (11506.0 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:13.711931: step 310, loss = 3.46 (11738.5 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:13.942842: step 320, loss = 3.49 (10844.9 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:14.173383: step 330, loss = 3.40 (11738.6 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:14.401010: step 340, loss = 3.40 (10915.0 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:14.625820: step 350, loss = 3.23 (11685.3 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:14.848921: step 360, loss = 3.45 (11440.2 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:15.081600: step 370, loss = 3.24 (11432.9 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:15.322751: step 380, loss = 3.31 (11381.4 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:15.560423: step 390, loss = 3.42 (10928.8 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:15.782916: step 400, loss = 3.37 (11375.5 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:16.414329: step 410, loss = 3.21 (11588.7 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:16.647128: step 420, loss = 3.21 (12052.5 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:16.879954: step 430, loss = 3.27 (11356.2 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:17.110543: step 440, loss = 3.14 (11712.6 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:17.333729: step 450, loss = 3.20 (11526.3 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:17.560134: step 460, loss = 3.27 (10882.4 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:17.789360: step 470, loss = 3.38 (10155.5 examples/sec; 0.025 sec/batch)
2018-04-25 13:29:18.016032: step 480, loss = 3.33 (11506.6 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:18.241547: step 490, loss = 3.20 (12811.3 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:18.463968: step 500, loss = 3.01 (13084.5 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:19.085662: step 510, loss = 3.02 (11080.0 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:19.318092: step 520, loss = 3.10 (10035.7 examples/sec; 0.026 sec/batch)
2018-04-25 13:29:19.548764: step 530, loss = 3.02 (13011.3 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:19.780945: step 540, loss = 3.18 (10964.3 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:20.010748: step 550, loss = 3.05 (11462.2 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:20.233549: step 560, loss = 3.02 (12045.3 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:20.460756: step 570, loss = 2.96 (11357.8 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:20.691234: step 580, loss = 2.87 (11361.5 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:20.925262: step 590, loss = 2.91 (10552.5 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:21.153429: step 600, loss = 2.83 (11256.3 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:21.784791: step 610, loss = 2.88 (10463.0 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:22.020176: step 620, loss = 2.95 (11266.1 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:22.245832: step 630, loss = 2.81 (10943.7 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:22.478603: step 640, loss = 3.06 (11315.1 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:22.699617: step 650, loss = 2.80 (11073.7 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:22.923298: step 660, loss = 2.72 (11142.4 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:23.138955: step 670, loss = 2.85 (12053.3 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:23.368193: step 680, loss = 2.83 (11551.3 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:23.593977: step 690, loss = 2.66 (11551.7 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:23.830126: step 700, loss = 2.78 (10838.4 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:24.451436: step 710, loss = 2.75 (10416.5 examples/sec; 0.025 sec/batch)
2018-04-25 13:29:24.681334: step 720, loss = 2.73 (11041.8 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:24.902077: step 730, loss = 2.72 (10872.5 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:25.136414: step 740, loss = 2.56 (9861.2 examples/sec; 0.026 sec/batch)
2018-04-25 13:29:25.370181: step 750, loss = 2.68 (10561.2 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:25.599532: step 760, loss = 2.63 (11065.7 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:25.821157: step 770, loss = 2.63 (11124.1 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:26.049117: step 780, loss = 3.12 (12820.8 examples/sec; 0.020 sec/batch)
2018-04-25 13:29:26.272191: step 790, loss = 2.80 (11133.2 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:26.503673: step 800, loss = 2.67 (10597.9 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:27.136798: step 810, loss = 2.58 (11127.2 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:27.364822: step 820, loss = 2.61 (10974.5 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:27.599396: step 830, loss = 2.66 (10411.4 examples/sec; 0.025 sec/batch)
2018-04-25 13:29:27.827089: step 840, loss = 2.64 (10649.8 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:28.056812: step 850, loss = 2.64 (11920.4 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:28.279118: step 860, loss = 2.43 (12043.4 examples/sec; 0.021 sec/batch)
2018-04-25 13:29:28.504483: step 870, loss = 2.50 (11692.3 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:28.736615: step 880, loss = 2.56 (11327.2 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:28.962184: step 890, loss = 2.45 (10845.5 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:29.192103: step 900, loss = 2.43 (10905.5 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:29.802085: step 910, loss = 2.58 (10948.4 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:30.028644: step 920, loss = 2.44 (10968.6 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:30.257076: step 930, loss = 2.31 (10759.7 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:30.487558: step 940, loss = 2.40 (11642.4 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:30.713845: step 950, loss = 2.52 (10281.8 examples/sec; 0.025 sec/batch)
2018-04-25 13:29:30.946923: step 960, loss = 2.36 (11419.3 examples/sec; 0.022 sec/batch)
2018-04-25 13:29:31.185377: step 970, loss = 2.41 (11133.0 examples/sec; 0.023 sec/batch)
2018-04-25 13:29:31.415116: step 980, loss = 2.30 (10664.0 examples/sec; 0.024 sec/batch)
2018-04-25 13:29:31.639376: step 990, loss = 2.44 (10769.7 examples/sec; 0.024 sec/batch)

New code logs:

python cifar10_main.py --data-dir=${PWD}/cifar-10-data --job-dir=/tmp/cifar10 --num-gpus=1 --train-steps=1000 --train-batch-size=256 --eval-batch-size=1000

2018-04-25 13:29:57.739155: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2018-04-25 13:29:58.228832: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties: 
name: Quadro P6000 major: 6 minor: 1 memoryClockRate(GHz): 1.645
pciBusID: 0000:03:00.0
totalMemory: 23.87GiB freeMemory: 23.70GiB
2018-04-25 13:29:58.681472: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 1 with properties: 
name: Quadro P400 major: 6 minor: 1 memoryClockRate(GHz): 1.2525
pciBusID: 0000:04:00.0
totalMemory: 1.94GiB freeMemory: 1.06GiB
2018-04-25 13:29:59.072632: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 2 with properties: 
name: Quadro P6000 major: 6 minor: 1 memoryClockRate(GHz): 1.645
pciBusID: 0000:84:00.0
totalMemory: 23.87GiB freeMemory: 23.70GiB
2018-04-25 13:29:59.072760: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Device peer to peer matrix
2018-04-25 13:29:59.072800: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1051] DMA: 0 1 2 
2018-04-25 13:29:59.072813: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 0:   Y N N 
2018-04-25 13:29:59.072821: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 1:   N Y N 
2018-04-25 13:29:59.072829: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 2:   N N Y 
2018-04-25 13:29:59.072858: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Quadro P6000, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-04-25 13:29:59.072870: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1107] Ignoring gpu device (device: 1, name: Quadro P400, pci bus id: 0000:04:00.0, compute capability: 6.1) with Cuda multiprocessor count: 2. The minimum required count is 8. You can adjust this requirement with the env var TF_MIN_GPU_MULTIPROCESSOR_COUNT.
2018-04-25 13:29:59.072881: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:1) -> (device: 2, name: Quadro P6000, pci bus id: 0000:84:00.0, compute capability: 6.1)
INFO:tensorflow:Using config: {'_task_type': None, '_task_id': 0, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7fae5edd52e8>, '_master': '', '_num_ps_replicas': 0, '_num_worker_replicas': 0, '_environment': 'local', '_is_chief': True, '_evaluation_master': '', '_tf_config': gpu_options {
  per_process_gpu_memory_fraction: 1.0
}
, '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_secs': 600, '_log_step_count_steps': 100, '_session_config': gpu_options {
  force_gpu_compatible: true
}
allow_soft_placement: true
, '_save_checkpoints_steps': None, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_model_dir': '/tmp/cifar10'}
WARNING:tensorflow:From /home/ahmed/anaconda3/envs/tensorflow/lib/python3.6/site-packages/tensorflow/contrib/learn/python/learn/monitors.py:267: BaseMonitor.__init__ (from tensorflow.contrib.learn.python.learn.monitors) is deprecated and will be removed after 2016-12-05.
Instructions for updating:
Monitors are deprecated. Please use tf.train.SessionRunHook.
WARNING:tensorflow:From /home/ahmed/Documents/tensorflow/models/tutorials/image/cifar10_estimator/cifar10.py:74: TFRecordDataset.__init__ (from tensorflow.contrib.data.python.ops.readers) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.TFRecordDataset`.
WARNING:tensorflow:From /home/ahmed/Documents/tensorflow/models/tutorials/image/cifar10_estimator/cifar10.py:78: calling Dataset.map (from tensorflow.contrib.data.python.ops.dataset_ops) with num_threads is deprecated and will be removed in a future version.
Instructions for updating:
Replace `num_threads=T` with `num_parallel_calls=T`. Replace `output_buffer_size=N` with `ds.prefetch(N)` on the returned dataset.
WARNING:tensorflow:From /home/ahmed/Documents/tensorflow/models/tutorials/image/cifar10_estimator/cifar10.py:78: calling Dataset.map (from tensorflow.contrib.data.python.ops.dataset_ops) with output_buffer_size is deprecated and will be removed in a future version.
Instructions for updating:
Replace `num_threads=T` with `num_parallel_calls=T`. Replace `output_buffer_size=N` with `ds.prefetch(N)` on the returned dataset.
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_1/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_2/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_3/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_4/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_5/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_6/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1/avg_pool/: (?, 16, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_1/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_2/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_3/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_4/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_5/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_6/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1/avg_pool/: (?, 32, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_1/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_2/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_3/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_4/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_5/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_6/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/global_avg_pool/: (?, 64)
INFO:tensorflow:image after unit resnet/tower_0/fully_connected/: (?, 11)
INFO:tensorflow:Create CheckpointSaverHook.
2018-04-25 13:30:04.001300: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Quadro P6000, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-04-25 13:30:04.001366: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1107] Ignoring gpu device (device: 1, name: Quadro P400, pci bus id: 0000:04:00.0, compute capability: 6.1) with Cuda multiprocessor count: 2. The minimum required count is 8. You can adjust this requirement with the env var TF_MIN_GPU_MULTIPROCESSOR_COUNT.
2018-04-25 13:30:04.001379: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:1) -> (device: 2, name: Quadro P6000, pci bus id: 0000:84:00.0, compute capability: 6.1)
INFO:tensorflow:Saving checkpoints for 1 into /tmp/cifar10/model.ckpt.
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_1/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_2/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_3/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_4/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_5/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_6/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1/avg_pool/: (?, 16, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_1/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_2/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_3/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_4/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_5/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_6/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1/avg_pool/: (?, 32, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_1/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_2/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_3/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_4/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_5/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_6/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/global_avg_pool/: (?, 64)
INFO:tensorflow:image after unit resnet/tower_0/fully_connected/: (?, 11)
INFO:tensorflow:Starting evaluation at 2018-04-25-19:30:13
2018-04-25 13:30:14.677920: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Quadro P6000, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-04-25 13:30:14.677967: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1107] Ignoring gpu device (device: 1, name: Quadro P400, pci bus id: 0000:04:00.0, compute capability: 6.1) with Cuda multiprocessor count: 2. The minimum required count is 8. You can adjust this requirement with the env var TF_MIN_GPU_MULTIPROCESSOR_COUNT.
2018-04-25 13:30:14.677977: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:1) -> (device: 2, name: Quadro P6000, pci bus id: 0000:84:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from /tmp/cifar10/model.ckpt-1
INFO:tensorflow:Evaluation [1/10]
INFO:tensorflow:Evaluation [2/10]
INFO:tensorflow:Evaluation [3/10]
INFO:tensorflow:Evaluation [4/10]
INFO:tensorflow:Evaluation [5/10]
INFO:tensorflow:Evaluation [6/10]
INFO:tensorflow:Evaluation [7/10]
INFO:tensorflow:Evaluation [8/10]
INFO:tensorflow:Evaluation [9/10]
INFO:tensorflow:Evaluation [10/10]
INFO:tensorflow:Finished evaluation at 2018-04-25-19:30:16
INFO:tensorflow:Saving dict for global step 1: accuracy = 0.1, global_step = 1, loss = 1889.2959
INFO:tensorflow:Validation (step 1): accuracy = 0.1, loss = 1889.2959, global_step = 1
INFO:tensorflow:loss = 6.9046836, step = 1
INFO:tensorflow:learning_rate = 0.1, loss = 6.9046836
INFO:tensorflow:Average examples/sec: 2017.66 (2017.66), step = 10
INFO:tensorflow:Average examples/sec: 2486.74 (3240), step = 20
INFO:tensorflow:Average examples/sec: 2673.58 (3146.41), step = 30
INFO:tensorflow:Average examples/sec: 2735.68 (2940.58), step = 40
INFO:tensorflow:Average examples/sec: 2791.57 (3040.02), step = 50
INFO:tensorflow:Average examples/sec: 2841.23 (3118.56), step = 60
INFO:tensorflow:Average examples/sec: 2885.23 (3180.8), step = 70
INFO:tensorflow:Average examples/sec: 2919.9 (3188.04), step = 80
INFO:tensorflow:Average examples/sec: 2947.37 (3187.28), step = 90
INFO:tensorflow:global_step/sec: 6.63042
INFO:tensorflow:loss = 2.8437786, step = 101 (8.626 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.8437786 (8.626 sec)
INFO:tensorflow:Average examples/sec: 2967.61 (3163.14), step = 100
INFO:tensorflow:Average examples/sec: 2977.61 (3081.45), step = 110
INFO:tensorflow:Average examples/sec: 2983.44 (3049.13), step = 120
INFO:tensorflow:Average examples/sec: 2995.74 (3151.59), step = 130
INFO:tensorflow:Average examples/sec: 3007.38 (3167.46), step = 140
INFO:tensorflow:Average examples/sec: 3019.48 (3199.67), step = 150
INFO:tensorflow:Average examples/sec: 3028.82 (3176.18), step = 160
INFO:tensorflow:Average examples/sec: 3038.65 (3205.01), step = 170
INFO:tensorflow:Average examples/sec: 3046.15 (3179.7), step = 180
INFO:tensorflow:Average examples/sec: 3046.98 (3062.01), step = 190
INFO:tensorflow:global_step/sec: 12.2695
INFO:tensorflow:loss = 2.8192415, step = 201 (8.150 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.8192415 (8.150 sec)
INFO:tensorflow:Average examples/sec: 3051.84 (3147.09), step = 200
INFO:tensorflow:Average examples/sec: 3057.61 (3177.79), step = 210
INFO:tensorflow:Average examples/sec: 3064.08 (3206.46), step = 220
INFO:tensorflow:Average examples/sec: 3069.99 (3206.1), step = 230
INFO:tensorflow:Average examples/sec: 3074.48 (3181.68), step = 240
INFO:tensorflow:Average examples/sec: 3077.84 (3160.68), step = 250
INFO:tensorflow:Average examples/sec: 3076.69 (3048.07), step = 260
INFO:tensorflow:Average examples/sec: 3079.83 (3163.96), step = 270
INFO:tensorflow:Average examples/sec: 3083.93 (3198.88), step = 280
INFO:tensorflow:Average examples/sec: 3087.72 (3197.82), step = 290
INFO:tensorflow:global_step/sec: 12.3913
INFO:tensorflow:loss = 2.7655356, step = 301 (8.070 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.7655356 (8.070 sec)
INFO:tensorflow:Average examples/sec: 3091 (3189.02), step = 300
INFO:tensorflow:Average examples/sec: 3094.57 (3205.77), step = 310
INFO:tensorflow:Average examples/sec: 3097.34 (3185.74), step = 320
INFO:tensorflow:Average examples/sec: 3096.22 (3060.77), step = 330
INFO:tensorflow:Average examples/sec: 3099.09 (3196.86), step = 340
INFO:tensorflow:Average examples/sec: 3102.16 (3210.19), step = 350
INFO:tensorflow:Average examples/sec: 3104.62 (3193.35), step = 360
INFO:tensorflow:Average examples/sec: 3107.4 (3210.78), step = 370
INFO:tensorflow:Average examples/sec: 3110.37 (3224.43), step = 380
INFO:tensorflow:Average examples/sec: 3112.97 (3215.14), step = 390
INFO:tensorflow:global_step/sec: 12.4225
INFO:tensorflow:loss = 2.731544, step = 401 (8.050 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.731544 (8.050 sec)
INFO:tensorflow:Average examples/sec: 3112.81 (3106.81), step = 400
INFO:tensorflow:Average examples/sec: 3114.35 (3176.98), step = 410
INFO:tensorflow:Average examples/sec: 3115.91 (3181.15), step = 420
INFO:tensorflow:Average examples/sec: 3117.7 (3194.77), step = 430
INFO:tensorflow:Average examples/sec: 3120.11 (3227.5), step = 440
INFO:tensorflow:Average examples/sec: 3122.44 (3228.72), step = 450
INFO:tensorflow:Average examples/sec: 3123.04 (3150.18), step = 460
INFO:tensorflow:Average examples/sec: 3123.92 (3164.84), step = 470
INFO:tensorflow:Average examples/sec: 3124.27 (3141.04), step = 480
INFO:tensorflow:Average examples/sec: 3125.6 (3190.45), step = 490
INFO:tensorflow:global_step/sec: 12.4265
INFO:tensorflow:loss = 2.4471266, step = 501 (8.047 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.4471266 (8.047 sec)
INFO:tensorflow:Average examples/sec: 3126.25 (3158.44), step = 500
INFO:tensorflow:Average examples/sec: 3127 (3165.09), step = 510
INFO:tensorflow:Average examples/sec: 3128.51 (3207.75), step = 520
INFO:tensorflow:Average examples/sec: 3128.94 (3151.16), step = 530
INFO:tensorflow:Average examples/sec: 3129.28 (3147.37), step = 540
INFO:tensorflow:Average examples/sec: 3130.13 (3177.18), step = 550
INFO:tensorflow:Average examples/sec: 3131.59 (3213.57), step = 560
INFO:tensorflow:Average examples/sec: 3132.75 (3199.17), step = 570
INFO:tensorflow:Average examples/sec: 3133.95 (3203.91), step = 580
INFO:tensorflow:Average examples/sec: 3135.28 (3214.47), step = 590
INFO:tensorflow:global_step/sec: 12.4362
INFO:tensorflow:loss = 2.261356, step = 601 (8.041 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.261356 (8.041 sec)
INFO:tensorflow:Average examples/sec: 3135.68 (3159.47), step = 600
INFO:tensorflow:Average examples/sec: 3134.24 (3050.07), step = 610
INFO:tensorflow:Average examples/sec: 3134.3 (3138.29), step = 620
INFO:tensorflow:Average examples/sec: 3135.32 (3199.53), step = 630
INFO:tensorflow:Average examples/sec: 3136.67 (3224.55), step = 640
INFO:tensorflow:Average examples/sec: 3137.64 (3200.59), step = 650
INFO:tensorflow:Average examples/sec: 3138.81 (3217.21), step = 660
INFO:tensorflow:Average examples/sec: 3139.29 (3170.85), step = 670
INFO:tensorflow:Average examples/sec: 3138.22 (3068.07), step = 680
INFO:tensorflow:Average examples/sec: 3138.49 (3157.52), step = 690
INFO:tensorflow:global_step/sec: 12.3327
INFO:tensorflow:loss = 2.1352093, step = 701 (8.108 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.1352093 (8.109 sec)
INFO:tensorflow:Average examples/sec: 3138.73 (3155.15), step = 700
INFO:tensorflow:Average examples/sec: 3139.42 (3188.24), step = 710
INFO:tensorflow:Average examples/sec: 3140.08 (3188.08), step = 720
INFO:tensorflow:Average examples/sec: 3141.02 (3210.13), step = 730
INFO:tensorflow:Average examples/sec: 3141.51 (3177.39), step = 740
INFO:tensorflow:Average examples/sec: 3140.97 (3101.53), step = 750
INFO:tensorflow:Average examples/sec: 3141.26 (3163.25), step = 760
INFO:tensorflow:Average examples/sec: 3141.43 (3154.55), step = 770
INFO:tensorflow:Average examples/sec: 3141.99 (3185.77), step = 780
INFO:tensorflow:Average examples/sec: 3142.79 (3206.68), step = 790
INFO:tensorflow:global_step/sec: 12.399
INFO:tensorflow:loss = 2.1100209, step = 801 (8.065 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.1100209 (8.065 sec)
INFO:tensorflow:Average examples/sec: 3143.12 (3168.93), step = 800
INFO:tensorflow:Average examples/sec: 3143.36 (3163.21), step = 810
INFO:tensorflow:Average examples/sec: 3143.08 (3120.26), step = 820
INFO:tensorflow:Average examples/sec: 3143.66 (3191.53), step = 830
INFO:tensorflow:Average examples/sec: 3144.03 (3175.14), step = 840
INFO:tensorflow:Average examples/sec: 3144.35 (3171.98), step = 850
INFO:tensorflow:Average examples/sec: 3145.12 (3211.61), step = 860
INFO:tensorflow:Average examples/sec: 3145.73 (3199.38), step = 870
INFO:tensorflow:Average examples/sec: 3145.81 (3152.73), step = 880
INFO:tensorflow:Average examples/sec: 3145.47 (3116.02), step = 890
INFO:tensorflow:global_step/sec: 12.348
INFO:tensorflow:loss = 2.0054061, step = 901 (8.098 sec)
INFO:tensorflow:learning_rate = 0.1, loss = 2.0054061 (8.098 sec)
INFO:tensorflow:Average examples/sec: 3145.11 (3112.75), step = 900
INFO:tensorflow:Average examples/sec: 3145.24 (3157.23), step = 910
INFO:tensorflow:Average examples/sec: 3145.57 (3176.37), step = 920
INFO:tensorflow:Average examples/sec: 3144.61 (3058.61), step = 930
INFO:tensorflow:Average examples/sec: 3143.99 (3087.54), step = 940
INFO:tensorflow:Average examples/sec: 3143.53 (3100.39), step = 950
INFO:tensorflow:Average examples/sec: 3142.7 (3065.62), step = 960
INFO:tensorflow:Average examples/sec: 3141.94 (3070.97), step = 970
INFO:tensorflow:Average examples/sec: 3141.43 (3092.79), step = 980
INFO:tensorflow:Average examples/sec: 3141.29 (3127.61), step = 990
INFO:tensorflow:Saving checkpoints for 1000 into /tmp/cifar10/model.ckpt.
INFO:tensorflow:Loss for final step: 1.8685846.
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_1/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_2/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_3/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_4/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_5/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage/residual_v1_6/: (?, 16, 32, 32)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1/avg_pool/: (?, 16, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_1/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_2/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_3/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_4/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_5/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_1/residual_v1_6/: (?, 32, 16, 16)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1/avg_pool/: (?, 32, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_1/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_2/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_3/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_4/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_5/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/stage_2/residual_v1_6/: (?, 64, 8, 8)
INFO:tensorflow:image after unit resnet/tower_0/global_avg_pool/: (?, 64)
INFO:tensorflow:image after unit resnet/tower_0/fully_connected/: (?, 11)
INFO:tensorflow:Starting evaluation at 2018-04-25-19:31:41
2018-04-25 13:31:42.977257: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Quadro P6000, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-04-25 13:31:42.977312: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1107] Ignoring gpu device (device: 1, name: Quadro P400, pci bus id: 0000:04:00.0, compute capability: 6.1) with Cuda multiprocessor count: 2. The minimum required count is 8. You can adjust this requirement with the env var TF_MIN_GPU_MULTIPROCESSOR_COUNT.
2018-04-25 13:31:42.977321: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:1) -> (device: 2, name: Quadro P6000, pci bus id: 0000:84:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from /tmp/cifar10/model.ckpt-1000
INFO:tensorflow:Evaluation [1/10]
INFO:tensorflow:Evaluation [2/10]
INFO:tensorflow:Evaluation [3/10]
INFO:tensorflow:Evaluation [4/10]
INFO:tensorflow:Evaluation [5/10]
INFO:tensorflow:Evaluation [6/10]
INFO:tensorflow:Evaluation [7/10]
INFO:tensorflow:Evaluation [8/10]
INFO:tensorflow:Evaluation [9/10]
INFO:tensorflow:Evaluation [10/10]
INFO:tensorflow:Finished evaluation at 2018-04-25-19:31:44
INFO:tensorflow:Saving dict for global step 1000: accuracy = 0.2721, global_step = 1000, loss = 2.4688354
